src/runtime/security/negotiation_service.cpp

```diff
@@ -67,12 +69,32 @@ void negotiation_service::on_negotiation_request(negotiation_rpc rpc)
             static_cast<server_negotiation *>(_negotiations[rpc.dsn_request()->io_session].get());
     }
 
-    dassert(srv_negotiation != nullptr,
-            "negotiation is null for msg: {}",
-            rpc.dsn_request()->rpc_code().to_string());
+    if (nullptr == srv_negotiation) {
+        derror_f("negotiation is null for msg: {}", rpc.dsn_request()->rpc_code().to_string());
+        return;
+    }
     srv_negotiation->handle_request(rpc);
 }
 
+void negotiation_service::on_negotiation_response(error_code err, negotiation_rpc rpc)
```

Comment:
The name `negotiation_service` is ambiguous. It conflicts with `server_negotiation` that easily gets people confused about the difference. Apart from adding an explanation in the comment, I recommend you rename the class, to "negotiation_manager" or so.

Reply:
done.